### PR TITLE
Generate applications without entities among a mixed list in JDL

### LINKED ISSUE #13141: JDL: an application is not generated if it doesn't contain entities
As @pmverma reported in our mailing list (thanks again).

##### **Overview of the issue and  Motivation for or Use Case**

An application should be generated whether it contains entities or not, it's not the case.

##### **Reproduce the error**

Importing this JDL sample only generates the first app.
```jdl
application {
  config {
    baseName app1
  }
  entities A
}

application {
  config {
    baseName app2
  }
}
```

##### **Related issues**

I don't think there is one.

##### **Suggest a Fix**

[This](https://github.com/jhipster/generator-jhipster/blob/main/cli/import-jdl.js#L389) seems to be the issue.
A test should be made for this case.

##### **JHipster Version(s)**

Tested on main branch.

-   [X] Checking this box is mandatory (this is just to show you read everything)
-   [X] Yes I do love JHipster.

### PR DESCRIPTION
**Pull request by swarajsaaj**
Fixes bug to include applications with or without entities from the
config file, where applications without entities was ignored from a list
of applications with atleast one application with entities

Fix #13141

---

Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by MathieuAA** [comment]
Something bothers me: the need to do this function. Are applications without entities filtered somewhere?
Furthermore, the test doesn't cover the entire flow: testing the JDL importer is also required

**Comment by swarajsaaj** [comment]
> 
> 
> Something bothers me: the need to do this function. Are applications without entities filtered somewhere?
> Furthermore, the test doesn't cover the entire flow: testing the JDL importer is also required

@MathieuAA The importState in `jdl-importer.js` is constructed from `entitiesPerApplicationMap` through `JDLWithApplicationsToJSONConverter` (see https://github.com/jhipster/generator-jhipster/blob/main/jdl/jdl-importer.js#L277)

The flow is such that this application-entity map is constructed from`JDLWithApplicationsToJSONConverter.convert`, so rather than filtering out, these applications were not at all included in map (see https://github.com/jhipster/generator-jhipster/blob/main/jdl/converters/jdl-to-json/jdl-with-applications-to-json-converter.js#L49)

**Comment by MathieuAA** [comment]
Once the test is done, I think we should see things more clearly. Moreover, a test for the JDL cli command would be nice too!

File: jdl/converters/jdl-to-json/jdl-with-applications-to-json-converter.js [hunk 1]
```diff
@@ -165,3 +166,13 @@ function formatEntitiesForEachApplication(entitiesForEachApplication) {
...
+
+function addApplicationsWithoutEntities(entitiesForEachApplication) {
+    const applicationsWithoutEntities = jdlObject
+        .getApplications()
+        .filter(jdlApplication => jdlApplication.getEntityNames().length === 0)
+        .map(jdlApplication => jdlApplication.getConfigurationOptionValue('baseName'));
+    if (applicationsWithoutEntities.length > 0) {
+        applicationsWithoutEntities.forEach(applicationName => entitiesForEachApplication.set(applicationName, []));
+    }
+}
```
Comment (MathieuAA) [inline jdl/converters/jdl-to-json/jdl-with-applications-to-json-converter.js:178]: if this is the solution, we could do it in a more optimized way:
  - loop through the applications (`getApplications()` then forEach)
  - in the forEach loop, check if there are assigned entities and immediately add them

**Comment by swarajsaaj** [comment]
> Once the test is done, I think we should see things more clearly. Moreover, a test for the JDL cli command would be nice too!

@MathieuAA  could you please explain what do you mean by  'JDL cli command' test, is it integration-test? or where are they located?

**Comment by MathieuAA** [comment]
In `test/cli/jdl/import-jdl.spec.js`

File: jdl/converters/jdl-to-json/jdl-with-applications-to-json-converter.js (hunk 1 above)
Comment (swarajsaaj) [reply jdl/converters/jdl-to-json/jdl-with-applications-to-json-converter.js]: Agreed, infact there is a `jdlObject.forEachApplication()` for iteration over applications.

**Comment by pascalgrimaud** [comment]
@MathieuAA : if you feel confident, can we merge this for v7 ?

**Comment by MathieuAA** [comment]
Yes, let's do this

**Review by MathieuAA — APPROVED** [review]

**Comment by swarajsaaj** [comment]
Bounty claim at https://opencollective.com/generator-jhipster/expenses/30284

**Comment by pascalgrimaud** [comment]
@swarajsaaj : approved

**Comment by MathieuAA** [comment]
@swarajsaaj great job :)

**Comment by swarajsaaj** [comment]
Thanks @MathieuAA  :)

**Pull request merged**